src/components/form/FormInput.tsx

```diff
@@ -1,5 +1,94 @@
 import React from 'react';
+import { UseFormRegister } from 'react-hook-form';
 
-export const FormInput = () => {
-  return <div>FormInput</div>;
+export interface IFormInputProps {
+  warning?: boolean;
+  success?: boolean;
+  statusText?: string;
+  icon?: JSX.Element;
+  label: string;
+  register?: UseFormRegister<any>;
+  type?: string;
+  disabled?: boolean;
+  id: string;
+}
+
+/**
+ *
+ * Email input form field component
+ *
+ * @param {boolean} warning - on validation failure
+ * @param {boolean} success - on validation success
+ * @param {string} statusText - validation status text
+ * @param {string} label - input label
+ *
+ * @returns {JSX.Element}
+ *
+ * @example
+ * <FormInput success />
+ *
+ */
+function FormInput({
+  warning,
+  success,
+  statusText,
+  label,
+  type,
+  icon,
+  register,
+  disabled,
+  id,
+}: IFormInputProps) {
+  return (
+    <div className="my-8">
+      <div className="relative h-10">
+        {icon && (
+          <span className="absolute top-1/2 left-0 z-20 ml-2 -translate-y-1/2">
+            {icon}
```

Comment:
I think you should add some more space between the icon and the input, i think it would look much better